src/modules/Resources/Buckets.ts

```diff
@@ -26,159 +24,71 @@ class Buckets extends TagoIOModule<GenericModuleParams> {
    * }
    * ```
    * @param queryObj Search query params
+   *
+   * @deprecated Use the method from `Resources.devices` instead.
    */
-  public async list(queryObj?: BucketQuery): Promise<BucketInfo[]> {
-    let result = await this.doRequest<BucketInfo[]>({
-      path: "/bucket",
-      method: "GET",
-      params: {
-        page: queryObj?.page || 1,
-        fields: queryObj?.fields || ["id", "name"],
-        filter: queryObj?.filter || {},
-        amount: queryObj?.amount || 20,
-        orderBy: queryObj?.orderBy ? `${queryObj.orderBy[0]},${queryObj.orderBy[1]}` : "name,asc",
-      },
-    });
-
-    result = result.map((data) => dateParser(data, ["created_at", "updated_at"]));
-
-    return result;
+  public async list<T extends DeviceQuery>(queryObj?: T) {
```

Comment:
All CRUD (including `list` and `amount`) is called from the `Devices` module, so the method signature was updated.

Did not set explicit return types to use the ones from the `Devices` module.